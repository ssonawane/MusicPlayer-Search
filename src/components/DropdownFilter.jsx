import React, { useState, useEffect } from 'react';
import Checkbox from './CheckBox';

function DropdownFilter({ getDrpDwnFilter }) {
    const defaultArr =
        [{ flag: false, value: 'all', text: 'All' },
        { flag: true, value: 'title', text: 'Title' },
        { flag: false, value: 'desc', text: 'Description' },
        { flag: false, value: 'keywrds', text: 'Keywords' }];

    const [checkedItems, setCheckedItems] = useState(defaultArr);


    const changeHandler = e => {
        const filteredArr = [...checkedItems];

        filteredArr.forEach(item => {
            if (e.target.value === 'all') {
                item.flag = e.target.checked;
            } else if (item.value === e.target.value) {
                item.flag = e.target.checked
            }
        })

        setCheckedItems(filteredArr)
    }

    useEffect(() => {
        getDrpDwnFilter(checkedItems);
    }, [checkedItems])

    return <div className="dropdown">
        <a href="#" className="dropbtn">
            Activities
        </a>
        <div className="dropdown-content">
            {
                checkedItems.map(item => <Checkbox key={item.value} changeHandler={changeHandler} {...item} />)
            }
        </div>
    </div>
}
export default DropdownFilter;