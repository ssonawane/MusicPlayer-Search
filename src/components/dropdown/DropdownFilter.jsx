import React, { useState, useEffect, useContext } from 'react';
import Checkbox from '../checkbox/CheckBox';
import { context } from '../../App';
import { CONSTANT } from '../../constant';

function DropdownFilter({ getDrpDwnFilter }) {
    const musicListArr = useContext(context);
    let defaultArr = [{ flag: false, value: 'all', text: 'All' }];

    let checkboxListArr = (musicListArr?.length > 0) ? Object.keys(musicListArr?.[0]) : [];

    checkboxListArr?.forEach(value => {
        CONSTANT.CHECKBOXES_LIST.forEach(item => {
            if (value === item) {
                defaultArr.push({
                    flag: false,
                    value: item.toLocaleLowerCase(),
                    text: item.charAt(0).toUpperCase() + item.substr(1)
                })
            }
        })
    })

    const [checkedItems, setCheckedItems] = useState(defaultArr);

    const changeHandler = e => {
        const filteredArr = [...checkedItems];
        let checkAllFlag = true;

        filteredArr.forEach(item => {
            if (e.target.value === 'all') {
                item.flag = e.target.checked;
            } else if (item.value === e.target.value) {
                item.flag = e.target.checked;
                filteredArr[0].flag = false;
            }
        })

        for (let i = 1; i < filteredArr.length; i++) {
            if (!filteredArr[i]?.flag) {
                checkAllFlag = false;
                break;
            }
        }

        if (checkAllFlag) {
            filteredArr[0].flag = true;
        }

        setCheckedItems(filteredArr)
    }

    useEffect(() => {
        getDrpDwnFilter(checkedItems);
    }, [checkedItems])

    return <div className="dropdown">
        <a href="#" data-testid="activity" className="dropbtn">
            Activities
        </a>
        <div data-testid="drpdwn-content" className="dropdown-content">
            {
                checkedItems.map(item => <Checkbox key={item.value} changeHandler={changeHandler} {...item} />)
            }
        </div>
    </div>
}
export default DropdownFilter;