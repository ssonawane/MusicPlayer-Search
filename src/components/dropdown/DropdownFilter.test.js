import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import DropdownFilter from "./DropdownFilter";

describe("DropdownFilter ", () => {
  test("renders learn react link", () => {
    const NavbarComponent = renderer.create(<DropdownFilter />).toJSON();
    expect(NavbarComponent).toMatchSnapshot();
  });

  test("check for searched string", () => {
    const getDrpDwnFilter = jest.fn();
    const { getByTestId } = render(
      <DropdownFilter getDrpDwnFilter={getDrpDwnFilter} />
    );

    expect(getByTestId("activity")).toHaveTextContent("Activities");
  });
});
