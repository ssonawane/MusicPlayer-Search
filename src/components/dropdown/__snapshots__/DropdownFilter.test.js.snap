// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownFilter  renders learn react link 1`] = `
<div
  className="dropdown"
>
  <a
    className="dropbtn"
    data-testid="activity"
    href="#"
  >
    Activities
  </a>
  <div
    className="dropdown-content"
    data-testid="drpdwn-content"
  >
    <div
      className="checkboxContainer"
    >
      <label
        className="container"
        data-testid="All"
      >
        All
        <input
          checked={false}
          data-testid="all"
          onChange={[Function]}
          type="checkbox"
          value="all"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <div
      className="checkboxContainer"
    >
      <label
        className="container"
        data-testid="Title"
      >
        Title
        <input
          checked={true}
          data-testid="title"
          onChange={[Function]}
          type="checkbox"
          value="title"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <div
      className="checkboxContainer"
    >
      <label
        className="container"
        data-testid="Description"
      >
        Description
        <input
          checked={false}
          data-testid="desc"
          onChange={[Function]}
          type="checkbox"
          value="desc"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <div
      className="checkboxContainer"
    >
      <label
        className="container"
        data-testid="Keywords"
      >
        Keywords
        <input
          checked={false}
          data-testid="keywrds"
          onChange={[Function]}
          type="checkbox"
          value="keywrds"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
  </div>
</div>
`;
